code/datums/components/onwear_mood.dm

```diff
@@ -0,0 +1,50 @@
+// Add to clothing to give the wearer a mood buff and a unique examine str
+
+/datum/component/onwear_mood
+	/// the event the wearer experiences
+	var/datum/mood_event/saved_event
+	/// examine string added to examine
+	var/examine_string
+
+/datum/component/onwear_mood/Initialize(clear_after, datum/mood_event/saved_event, examine_string)
+
+	if(!isitem(parent))
+		return COMPONENT_INCOMPATIBLE
+
+	if(istype(saved_event))
+		src.saved_event = saved_event
+
+	src.examine_string = examine_string
+
+	if(isnum(clear_after))
+		QDEL_IN(src, clear_after)
```

Comment:
Actually, I don't think this should exist at all and you should instead just call QDEL_IN on the result of AddComponent itself. This doesn't look like it's really part of the component.

Reply:
how/where would i do that?

Reply:
I believe AddComponent returns the component it just created, so you can `QDEL_IN(AddComponent(etc), 2 MINUTES)`